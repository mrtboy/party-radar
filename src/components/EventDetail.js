import React from 'react';
import { connect } from 'react-redux';
import moment from 'moment';
import Map from './Map';


const EventDetail = (props) => {
  return (
    <div>
      <div className="list-item">
        <h2 className="list-item__title">Description:</h2>
        <h3>{props.event.description}</h3>
      </div>
      <div className="list-item">
        <h2 className="list-item__title">Start At:</h2>
        <h3>{moment(props.event.startDate).format('MMMM Do, YYYY')}</h3>
      </div>
      <div className="list-item">
        <h2 className="list-item__title">End At:</h2>
        <h3>{moment(props.event.endDate).format('MMMM Do, YYYY')}</h3>
      </div>
      <div className="list-item">
        <h2 className="list-item__title">Type of Event:</h2>
        <h3>{props.event.type}</h3>
      </div>
      <div className="list-item">
        <h2 className="list-item__title">Location:</h2>
        <h3>{props.event.locationName}</h3>
      </div>
      <div className="list-item">
        <h2 className="list-item__title">Address:</h2>
        <h3>{props.event.address}</h3>
      </div>
      <Map event={props.event} />
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    map: state.map
  };
};


export default connect(mapStateToProps)(EventDetail);