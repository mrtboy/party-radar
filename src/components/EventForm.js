import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';
import Map from './Map';
import { connect } from 'react-redux';

const now = moment();


class EventForm extends React.Component {
  constructor(props) {
    super(props);



    this.state = {
      title: props.event ? props.event.title : '',
      eventDate: props.event ? moment(props.event.eventDate) : moment(),
      startTime: props.event ? props.event.startTime : moment(),
      endTime: props.event ? props.event.endTime : moment(),
      type: props.event ? props.event.type : '',
      description: props.event ? props.event.description : '',
      themeClothes: props.event ? props.event.themeClothes : '',
      place_id: props.event ? props.event.place_id : '',
      address: props.event ? props.event.address : '',
      locationName: props.event ? props.event.locationName : '',
      geolocation: props.event ? props.event.geolocation : {},
      calendarFocused: null
    };
  }

  //#region  OnChange Value
  onTitleChange = (e) => {
    const title = e.target.value;
    this.setState(() => ({ title }));
  };
  onTypeChange = (e) => {
    const type = e.target.value;
    this.setState(() => ({ type }));
  };

  onDescriptionChange = (e) => {
    const description = e.target.value;
    this.setState(() => ({ description }));
  };

  onThemeChange = (e) => {
    const themeClothes = e.target.value;
    this.setState(() => ({ themeClothes }));
  };

  onStreeChange = (e) => {
    const street = e.target.value;
    this.setState(() => ({ street }));
  };

  onStreeNumberChange = (e) => {
    const streetNo = e.target.value;
    this.setState(() => ({ streetNo }));
  };

  onPlzChange = (e) => {
    const plz = e.target.value;
    this.setState(() => ({ plz }));
  };

  onCityChange = (e) => {
    const city = e.target.value;
    this.setState(() => ({ city }));
  };

  onCountryChange = (e) => {
    const country = e.target.value;
    this.setState(() => ({ country }));
  };

  onDateChange = (eventDate) => {
    if (eventDate) {
      this.setState(() => ({ eventDate }));
    }

  };
  onFocusChange = ({ focused }) => {
    this.setState(() => ({ calendarFocused: focused }));
  };

  onAddressChange = (e) => {
    const address = e.target.value;
    this.setState(() => ({ address }));
  }
 //#endregion

  onSubmit = (e) => {
    e.preventDefault();
    if (!this.state.title && !!this.state.geolocation.lat ) {
      this.setState(() => ({
        error: 'Please provide Title of party and real location.'
      }))
    } else {
      this.setState(() => ({ error: '' }))
      this.props.onSubmit({
        title: this.state.title,
        eventDate: this.state.eventDate.valueOf(),
        startTime: this.state.startTime.valueOf(),
        endTime: this.state.endTime.valueOf(),
        type: this.state.type,
        description: this.state.description,
        themeClothes: this.state.themeClothes,
        place_id: this.props.map.place_id,
        address: this.props.map.address,
        locationName: this.props.map.locationName,
        geolocation: this.props.map.geolocation,
      })
    }
  }


  render() {
    return (
      <div className="content-container">
        <form onSubmit={this.onSubmit} className="form">
          <input className="text-input" placeholder="Party Title" type="text" onChange={this.onTitleChange} value={this.state.title} />
                 <SingleDatePicker
          date={this.state.eventDate}
          onDateChange={this.onDateChange}
          focused={this.state.calendarFocused}
          onFocusChange={this.onFocusChange}
          numberOfMonths={1}
          isOutsideRange={() => false}//chech days availability
        />

          <input type="text" className="text-input" placeholder="Type of Party" onChange={this.onTypeChange} value={this.state.type} />
          <textarea type="text" className="text-area" placeholder="Description" onChange={this.onDescriptionChange} value={this.state.description} />
          <input type="text" className="text-input" placeholder="Theme Clothes" onChange={this.onThemeChange} value={this.state.themeClothes} />
          <input type="text" className="text-input" placeholder="Location Name" value={this.props.map.locationName} disabled />
  
          <div>
            {!!this.state.title && !!this.props.map.locationName ? <button className="button-add" > Save Event </button> :
                <p>Please Enter Title and Address from map for Event.</p>}
          </div>
        </form>
        <Map event={this.props.event} autoComplete={true}/>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  map: state.map
});


export default connect(mapStateToProps)(EventForm);
