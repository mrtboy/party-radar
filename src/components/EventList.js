import React from 'react';
import { connect } from 'react-redux';
import EventListItem from './EventListItem';
import getVisibleEvents from '../selectors/events';

export const EventList = (props) => (
  <div className="content-container">
    <div className="list-header">
      <h3>Title</h3>
      <h3>Place</h3>
    </div>
    <div  className="list-body">

    {props.events.length === 0 ? (
      <div className="list-item list-item--message">
      <p>No Event</p>
      </div>
    ): (
      props.events.map((event) =>
        <EventListItem key={event.id} {...event} />
      
      ))
    }
    </div>
  </div>
);


const mapStateToProps = (state) => {
  return {
    events: getVisibleEvents(state.events, state.filters)
  };
};

export default connect(mapStateToProps)(EventList);