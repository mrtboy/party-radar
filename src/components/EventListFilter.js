import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import { setTextFilter, sortByDate, setStartDate, setEndDate, setRange } from '../actions/filters';
import Map from './Map';
import { getLocation } from './../actions/map';

class EventListFilters extends React.Component {
  constructor(props){
    super(props);
    this.state = {
      calendarFocused: null,
      range: 0
    };

  }
  
  onDatesChange = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };
  onFocusChange = (calendarFocused) => {
    this.setState(() => ({ calendarFocused }));
  }
  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
  };
  onSortChange = (e) => {
    if (e.target.value === 'date') {
      this.props.sortByDate();
    }
  };

  handleRangeChange = (e) => {
    this.props.setRange(e.target.value, this.props.map.geolocation);
  }

  render() {
    return (
      <div className="content-container">
        <div className="input-group">
          <div className="input-group__item">
            <input
              type="text"
              className="text-input"
              placeholder="Search Events"
              value={this.props.filters.text}
              onChange={this.onTextChange}
            />
          </div>
          <div className="input-group__item">
            {/* <select
              className="select"
              value={this.props.filters.sortBy}
              onChange={this.onSortChange}
            >
              <option value="date">Date</option>
            </select> */}
          </div>
          <div className="input-group__item">
            <DateRangePicker
              startDate={this.props.filters.startDate}
              endDate={this.props.filters.endDate}
              onDatesChange={this.onDatesChange}
              focusedInput={this.state.calendarFocused}
              onFocusChange={this.onFocusChange}
              showClearDates={true}
              numberOfMonths={1}
              isOutsideRange={() => false}
            />
          </div>
          <div className="input-group__item">
            <input className="text-input" type="text" placeholder="Km" onChange={this.handleRangeChange} value={this.props.filters.range} />
          </div>
          
        </div>
        <div className="hiddenMap"><Map /></div>
        
      </div>
    );
  }
}


const mapStateToProps = (state) => ({
  filters: state.filters,
  map: state.map
});

const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (text) => dispatch(setTextFilter(text)),
  sortByDate: () => dispatch(sortByDate()),
  setStartDate: (startDate) => dispatch(setStartDate(startDate)),
  setEndDate: (endDate) => dispatch(setEndDate(endDate)),
  setRange: (range, currentLocation) => dispatch(setRange(range, currentLocation))
});
export default connect(mapStateToProps, mapDispatchToProps)(EventListFilters);

