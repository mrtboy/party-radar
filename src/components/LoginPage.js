import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">Party Radar</h1>
      <p>Check live party around yourself!</p>
      <img src="./images/google.png" className="btn--google" alt="Login with Google" onClick={startLogin}/>
    </div>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
