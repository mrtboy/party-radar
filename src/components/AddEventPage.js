import React from 'react';
import { connect } from 'react-redux';
import EventForm from './EventForm';
import {addEvent} from '../actions/events';

export class AddEventPage extends React.Component {
  onSubmit = (event) => {
    this.props.addEvent(event);
    this.props.history.push('/');
  }
  render() {
    return (
      <div className="content-container">
        <h1 className="list-header">Add New Event</h1>
        <EventForm onSubmit={this.onSubmit}/>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  addEvent: (event) => dispatch(addEvent(event))
});

export default connect(undefined, mapDispatchToProps)(AddEventPage);