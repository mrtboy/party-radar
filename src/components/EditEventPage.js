import React from 'react';
import { connect } from 'react-redux';
import EventForm from './EventForm';
import { editEventFromdb, removeEventFromdb } from '../actions/events';


export class EditEventPage extends React.Component {
  onSubmit = (event) => {
    this.props.editEventFromdb(this.props.event.id, event);
    this.props.history.push('/');
  };
  onRemove = () => {
    this.props.removeEventFromdb({ id: this.props.event.id });
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Event</h1>
          </div>
        </div>
        <div className="content-container">
          <EventForm
            event={this.props.event}
            onSubmit={this.onSubmit}
          />
          <button className="button button--secondary" onClick={this.onRemove}>Remove Event</button>
        </div>
      </div>
    );
  }
};

const mapStateToProps = ( state, props ) => {
  return {
    event: state.events.find((event) => event.id === props.match.params.id)
  };
};


const mapDispatchToProps = (dispatch, props) => ({
  editEventFromdb: (id, event) => dispatch(editEventFromdb(id, event)),
  removeEventFromdb: (data) => dispatch(removeEventFromdb(data))
});

export default connect(mapStateToProps,mapDispatchToProps)(EditEventPage);