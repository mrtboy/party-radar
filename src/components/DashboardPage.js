import React from 'react';
import EventListFilter from './EventListFilter';
import EventList  from './EventList';

const DashboardPage = () => (
  <div>
    <EventListFilter />
    <EventList />
  </div>
);

export default DashboardPage;
