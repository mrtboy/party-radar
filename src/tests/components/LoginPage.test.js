import React from 'react';
import { shallow } from 'enzyme';
import {LoginPage} from '../../components/LoginPage';
;
test('should render Login Page', ()=> {
  const wrapper = shallow(<LoginPage startLogin={()=>{}}/>);
  expect(wrapper).toMatchSnapshot();
});



test('Should call startLogin on button Click', ()=>{
  const startLogin = jest.fn();
  const wrapper = shallow(<LoginPage startLogin={startLogin}/>);
  wrapper.find('button').simulate('click');
  expect(startLogin).toHaveBeenCalled();
});