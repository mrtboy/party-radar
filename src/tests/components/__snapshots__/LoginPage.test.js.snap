// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Login Page 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Party Radar
    </h1>
    <p>
      Check live party around yourself!
    </p>
    <img
      alt="Login with Google"
      className="btn--google"
      onClick={[Function]}
      src="./images/google.png"
      srcset=""
    />
  </div>
</div>
`;
